### PR TITLE
Added erb view that lists custom emails

### PR DESCRIPTION
**Pull request by nrs1729**
#### Describe the changes you have made in this PR:
I added a view to list the custom emails created so far (the view is only visible to admins). 

### Screenshots of the changes (If any):
image
image

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@
     resources :assignments
   end
 
-  resources :custom_mails, only: [:new, :create, :edit, :show, :update]
+  resources :custom_mails, only: [:index, :new, :create, :edit, :show, :update]
```
[the remark this answers is not in the document]
Comment (ankitkataria) [reply config/routes.rb]: Can we fix this error?

File: app/views/custom_mails/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+                <tr>
+                  <td><%= mail.subject %></td>
+                  <td><%= mail.sent ? "Yes" : "No" %></td>
+                  <td><%= link_to 'Show', mail %></td>
+                  <td><%= link_to 'Edit', edit_custom_mail_url(mail) %></td>
+                </tr>
+            <% end %>
+            </tbody>
+          </table>
+        <% end %>
```
Comment (ankitkataria) [inline app/views/custom_mails/index.html.erb]: What if the mail count is 0? Could we have "No custom mails so far" is that case?

**Comment by ankitkataria** [comment]
Also rebase this PR to latest master, only the commits relevant to the issue should be present in the branch.

**Comment by nrs1729** [comment]
Sorry about the commits, fixed it. I added a case for if there are no emails and a test that checks there's a response and that it contains the mail subject, is that okay? Not sure if there's something else to cover for the coverage/coveralls check. The string literals violations hound is bringing up seem in line with the rest of the code?

File: spec/controllers/custom_mail_controller_spec.rb [hunk 3]
```diff
@@ -68,6 +68,14 @@
         expect(response.body).to eq("The mails were queued for sending!")
       end
     end
+
+    describe "#index" do
+      it "shows the list of custom views" do
+        get custom_mail_path(@mail)
```
Comment (ankitkataria) [inline spec/controllers/custom_mail_controller_spec.rb]: You'll have to do `get custom_mails_path` to actually check the index route, this should check for the index route. 
This is also the reason why the test coverage is failing.

Comment (nrs1729) [reply spec/controllers/custom_mail_controller_spec.rb]: Thanks, should be fixed now

**Pull request merged**